[dumb] New port

FtZPetruska opened the pull request:
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [X] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [X] The versioning scheme in `vcpkg.json` matches what upstream says.
- [X] The license declaration in `vcpkg.json` matches what upstream says. [1]
- [X] The installed as the "copyright" file matches what upstream says.
- [X] The source code of the component installed comes from an authoritative source. [2]
- [X] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is in the new port's versions file.
- [X] Only one version is added to each modified port's versions file.

[1] The license provided by upstream does not seem to fit any SPDX license name, so the field in `vcpkg.json` is left to `null`. The actual license file is still installed.
[2] While the initial DUMB project is no longer maintained, the original maintainer have encouraged using this fork (see the latest news on https://dumb.sourceforge.net/). According to [repology.org](https://repology.org/project/dumb/versions), this fork is shipped on Arch Linux, Gentoo, Homebrew, MSYS2, and others.

---

DUMB can be used to parse the IT, XM, S3M, and MOD formats. The `allegro5` port can be built with dumb to support these formats.

This port includes an unofficial CMake target since upstream does not export any. Additionally, since no symbols are exported, the port is marked as `ONLY_STATIC_LIBRARY`.

I was able to build successfully and link an example program on:
- x64-windows
- x64-windows-static
- x86-windows
- arm64-windows
- x64-uwp
- x64-linux
- x64-osx
- arm64-osx
- x64-mingw-static
- x64-mingw-dynamic

FtZPetruska committed:
[dumb] New port.

FtZPetruska committed:
[dumb] Add to baseline.

Cheney-W commented:
```
find_package(unofficial-dumb CONFIG REQUIRED)
target_link_libraries(main PRIVATE unofficial::dumb::dumb
```
Usage test passed with x64-windows triplet.

dg0yt commented on ports/dumb/portfile.cmake:
```diff
@@ -0,0 +1,47 @@
+vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
```
This should be limited to Windows if it is because of dllexport.

dg0yt commented on ports/dumb/portfile.cmake:
```diff
@@ -0,0 +1,47 @@
+vcpkg_fixup_pkgconfig()
+vcpkg_copy_pdbs()
+
+vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/lib/pkgconfig/dumb.pc" "libdumb" "dumb")
+if(NOT VCPKG_BUILD_TYPE)
+    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/dumb.pc" "libdumb" "dumbd")
+endif()
```
I would avoid unofficial config if there is pkg-config.
(But of course it can be suggested upstream.)

dg0yt commented on ports/dumb/portfile.cmake:
```diff
@@ -0,0 +1,47 @@
+endif()
+
+if(VCPKG_TARGET_IS_WINDOWS)
+    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/lib/pkgconfig/dumb.pc" "-lm" "")
+    if(NOT VCPKG_BUILD_TYPE)
+        vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/dumb.pc" "-lm" "")
+    endif()
```
I would at a space in front of the pattern, at least as a good example ;-)  
`" -lm"`

FtZPetruska committed:
[dumb] Apply suggestion from review.

- Remove unofficial config since a pkg-config file is provided.
- Update the match string for pkg-config fixes.
- Limit ONLY_STATIC_LIBRARY to Windows.

FtZPetruska committed:
[dumb] Update version.

FtZPetruska commented:
Thank you for the feedback!

I went ahead and removed the unofficial config since the upstream project provides a pc file and limited `ONLY_STATIC_LIBRARY` to Windows since it is indeed because of dllexport.

Cheney-W commented on ports/dumb/vcpkg.json:
```diff
@@ -0,0 +1,17 @@
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
+      "host": true
```
```suggestion
```
Please remove this dependence because the vcpkg_cmake_config_fixup() function is no longer used.

FtZPetruska replied:
Ah, good catch, thank you.

Removed the dependency on `vcpkg-cmake-config` in [f6418e5](https://github.com/microsoft/vcpkg/pull/30740/commits/f6418e5248c127ba57f60e3e9e341372d5aec17a).

FtZPetruska committed:
[dumb] Remove vcpkg-cmake-config dependency.

This dependency is no longer needed since the unofficial config was
removed.

FtZPetruska committed:
[dumb] Update version.

BillyONeal commented:
Awesome, thanks for the new port! (I look forward to filling out policheck exceptions :P)
